interpreter/cling/lib/Interpreter/CIFactory.cpp

```diff
@@ -1332,8 +1334,11 @@ static void stringifyPreprocSetting(PreprocessorOptions& PPOpts,
     // The compiler invocation is the owner of the diagnostic options.
     // Everything else points to them.
     DiagnosticOptions& DiagOpts = InvocationPtr->getDiagnosticOpts();
-    llvm::IntrusiveRefCntPtr<DiagnosticsEngine> Diags =
-        SetupDiagnostics(DiagOpts);
+    // add prefix to diagnostic messages if second compiler instance is existing
+    // e.g. in CUDA mode
+    llvm::IntrusiveRefCntPtr<DiagnosticsEngine> Diags = SetupDiagnostics(
+        DiagOpts,
+        COpts.CUDAHost ? (COpts.CUDADevice ? "cling-ptx" : "cling") : "");
```

Comment:
I find this hard to read and prefer
```suggestion
    std::string ExeName;
    if (COpts.CUDAHost)
      ExeName = COpts.CUDADevice ? "cling-ptx" : "cling";
    llvm::IntrusiveRefCntPtr<DiagnosticsEngine> Diags = SetupDiagnostics(
        DiagOpts, ExeName);
```